src/test/kotlin/org/opensearch/observability/validation/ValidatorTests.kt

```diff
@@ -0,0 +1,89 @@
+package org.opensearch.observability.validation
+
+import com.fasterxml.jackson.core.JsonParseException
+import com.worldturner.medeia.api.ValidationFailedException
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.opensearch.common.Strings
+import org.opensearch.common.xcontent.json.JsonXContent
+import java.io.File
+
+internal class ValidatorTests {
+    private fun buildIntegration(with: Map<String, Any> = mapOf(), without: Set<String> = setOf()): String {
+        val defaults = mapOf(
+            Pair("name", "test"),
+            Pair("description", "This is a test integration config"),
+            Pair("identification", "12345"),
+            Pair("categories", listOf("cat1", "cat2")),
+            Pair("collection", emptyList<String>()),
+            Pair("repository", mapOf(Pair("url", "https://example.com/"))),
+            Pair(
+                "version",
+                mapOf(
+                    Pair("integration", "0.0.1"),
+                    Pair("schema", "0.0.1"),
+                    Pair("resource", "0.0.1"),
+                )
+            )
+        )
+        val builder = JsonXContent.contentBuilder()
+        builder.startObject()
+        for (entry in defaults.entries) {
+            when {
+                without.contains(entry.key) -> continue
+                with.contains(entry.key) -> builder.field(entry.key, with[entry.key])
+                else -> builder.field(entry.key, entry.value)
+            }
+        }
+        builder.endObject()
+        return Strings.toString(builder)
+    }
+
+    @Test
+    fun testConfigInvalidJson() {
+        val config = "{"
+        val validator = Validator(IntegrationComponent.INTEGRATION)
+        assertThrows<JsonParseException> {
+            validator.validate(config).getOrThrow()
+        }
+    }
+
+    @Test
+    fun testValidatorSampleIntegrations() {

```

Comment:
if are you expecting valid response here - the name should reflect this
also separate each schema validation to a different test
add per schema test different invalid validation tests with different cases

Reply:
Split up the tests across methods, but I'm wondering how thorough we should be with per-schema checks. At some point that's just testing the validation library instead of our code. The main things to check here are that the paths are configured correctly, and that it's actually calling the validator, right?